### PR TITLE
[clang] Implement constexpr support for __builtin_popcountg

### PR DESCRIPTION
**Pull request by overmighty**

### CODE REVIEW

**Review by Sirraide — APPROVED** [review]

**Comment by Sirraide** [comment]
This needs a release note, but LGTM otherwise.

File: clang/docs/ReleaseNotes.rst
Comment (Sirraide) [inline clang/docs/ReleaseNotes.rst:161]:
```suggestion
- ``__builtin_popcountg``, a type-generic alternative to
  ``__builtin_popcount{,l,ll}`` with support for any unsigned integer type, is
  now constexpr and may be used in constant expressions.
```

**Review by Sirraide — CHANGES_REQUESTED** [review]

File: clang/docs/ReleaseNotes.rst
Comment (Sirraide) [reply clang/docs/ReleaseNotes.rst:161]: `__builtin_popcountg` already existed; the change is that it’s now constexpr.

Comment (overmighty) [reply clang/docs/ReleaseNotes.rst:161]: There was no release note for `__builtin_popcountg` at all yet. See https://github.com/llvm/llvm-project/pull/82359.

Comment (Sirraide) [reply clang/docs/ReleaseNotes.rst:161]: I see how it is. In that case I’d mention both that it was added and that it’s constexpr.

**Review by Sirraide — APPROVED** [review]

**Comment by Sirraide** [comment]
I’ll wait for CI to finish building the docs and then merge this; it was passing before, so I don’t think we need to wait for it again as the only thing that has changed since then is the release notes.

**Comment by Sirraide** [comment]
I see the name of one of the arrays was wrong; we really should just refactor this entire test to just use `static_assert`; not in this pr, though.

**Comment by overmighty** [comment]
Yes, I had actually caught the mistake but somehow reintroduced it at some point.

**Review by Sirraide — APPROVED** [review]

**Comment by overmighty** [comment]
Usage of `__int128` in the test broke the clang-armv8-quick build: https://lab.llvm.org/buildbot/#/builders/245/builds/21611/steps/5/logs/FAIL__Clang__constant-builtins-2_c.

**Pull request merged**
